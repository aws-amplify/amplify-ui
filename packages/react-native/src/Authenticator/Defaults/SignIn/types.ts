import { StyleProp, ViewStyle } from 'react-native';

import { AuthenticatorComponentDefaults } from '@aws-amplify/ui-react-core';
import { FormHandlers } from '../../types';

export interface SignInStyle {
  container: StyleProp<ViewStyle>;
}

export type SignInComponent = AuthenticatorComponentDefaults<
  FormHandlers & { style?: SignInStyle }
>['SignIn'];
