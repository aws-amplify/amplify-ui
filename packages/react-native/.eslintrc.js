module.exports = {
  ignorePatterns: ['dist', '.eslintrc.js', '*.config.js', 'jest.setup.js'],
  extends: ['@aws-amplify/amplify-ui/react'],
  overrides: [
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
  // point to local tsconfig
  parserOptions: { project: ['tsconfig.json'], tsconfigRootDir: __dirname },
};
