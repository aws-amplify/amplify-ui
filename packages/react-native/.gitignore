dist
.watchmanconfig

# ignore jest generated coverage directory
coverage 
