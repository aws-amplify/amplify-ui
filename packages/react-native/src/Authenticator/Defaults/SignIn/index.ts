export { default as SignIn } from './SignIn';
