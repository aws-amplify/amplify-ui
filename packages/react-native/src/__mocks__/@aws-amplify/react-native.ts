export default {
  loadUrlPolyfill: jest.fn(),
};
