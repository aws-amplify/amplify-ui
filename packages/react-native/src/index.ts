export {
  Authenticator,
  AuthenticatorProps,
  useAuthenticator,
  UseAuthenticator,
  withAuthenticator,
  WithAuthenticatorOptions,
} from './Authenticator';
export * from './InAppMessaging';
export {
  Theme,
  ThemeProvider,
  defaultDarkModeOverride,
  defaultTheme,
  useTheme,
} from './theme';
