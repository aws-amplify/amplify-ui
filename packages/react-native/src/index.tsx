import React from 'react';
import { Text, View } from 'react-native';

export function HelloWorld(): JSX.Element {
  return (
    <View>
      <Text>Hello World!</Text>
    </View>
  );
}
