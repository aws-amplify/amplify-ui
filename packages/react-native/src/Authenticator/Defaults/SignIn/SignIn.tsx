import React, { useMemo } from 'react';
import { authenticatorTextUtil } from '@aws-amplify/ui';

import {
  DefaultFooter,
  DefaultTextFormFields,
  DefaultHeader,
  DefaultContent,
} from '../../common';
import { useFieldValues } from '../../hooks';

import { DefaultSignInProps } from '../types';

const COMPONENT_NAME = 'SignIn';

const SignIn = ({
  fields,
  handleBlur,
  handleChange,
  handleSubmit,
  hideSignUp,
  toResetPassword,
  toSignUp,
  validationErrors,
  ...rest
}: DefaultSignInProps): JSX.Element => {
  const {
    getSignInTabText,
    getSignInText,
    getSignUpTabText,
    getForgotPasswordText,
  } = authenticatorTextUtil;

  const {
    disableFormSubmit: disabled,
    fields: fieldsWithHandlers,
    fieldValidationErrors,
    handleFormSubmit,
  } = useFieldValues({
    componentName: COMPONENT_NAME,
    fields,
    handleBlur,
    handleChange,
    handleSubmit,
    validationErrors,
  });

  const headerText = getSignInTabText();
  const forgotPasswordText = getForgotPasswordText(true);
  const signInText = getSignInText();
  const signUpText = getSignUpTabText();

  const buttons = useMemo(() => {
    const forgotPassword = {
      children: forgotPasswordText,
      onPress: toResetPassword,
    };
    return {
      primary: { children: signInText, disabled, onPress: handleFormSubmit },
      links: hideSignUp
        ? [forgotPassword]
        : [forgotPassword, { children: signUpText, onPress: toSignUp }],
    };
  }, [
    disabled,
    forgotPasswordText,
    handleFormSubmit,
    hideSignUp,
    signInText,
    signUpText,
    toResetPassword,
    toSignUp,
  ]);

  return (
    <DefaultContent
      {...rest}
      buttons={buttons}
      fields={fieldsWithHandlers}
      headerText={headerText}
      validationErrors={fieldValidationErrors}
    />
  );
};

SignIn.Footer = DefaultFooter;
SignIn.FormFields = DefaultTextFormFields;
SignIn.Header = DefaultHeader;

SignIn.displayName = COMPONENT_NAME;
export default SignIn;
