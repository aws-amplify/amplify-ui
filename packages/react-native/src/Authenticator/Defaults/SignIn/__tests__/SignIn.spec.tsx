import React from 'react';
import { render } from '@testing-library/react-native';

import { authenticatorTextUtil } from '@aws-amplify/ui';
import SignIn from '../SignIn';

const { getSignInText, getForgotPasswordText } = authenticatorTextUtil;

const username = {
  name: 'username',
  label: 'Username',
  placeholder: 'Username',
  type: 'default' as const,
};

const password = {
  name: 'password',
  label: 'Password',
  placeholder: 'Password',
  type: 'password' as const,
};

const fields = [username, password];

const props = {
  fields,
  Footer: SignIn.Footer,
  FormFields: SignIn.FormFields,
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  hasValidationErrors: false,
  Header: SignIn.Header,
  isPending: false,
  socialProviders: undefined,
  toResetPassword: jest.fn(),
  toFederatedSignIn: jest.fn(),
  toSignUp: jest.fn(),
};

describe('SignIn', () => {
  it('renders as expected', () => {
    const { toJSON, getAllByRole, getByText } = render(<SignIn {...props} />);
    expect(toJSON()).toMatchSnapshot();

    expect(getAllByRole('text')).toHaveLength(fields.length);
    expect(getByText(getSignInText())).toBeDefined();
    expect(getByText(getForgotPasswordText(true))).toBeDefined();
  });

  it('renders as expected with an error', () => {
    const error = 'An error!';
    const { getByRole, getByText, toJSON } = render(
      <SignIn {...props} error={error} />
    );

    expect(getByRole('alert')).toBeDefined();
    expect(getByText(error)).toBeDefined();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders as expected when hideSignUp is true', () => {
    const { toJSON } = render(<SignIn {...props} hideSignUp />);
    expect(toJSON()).toMatchSnapshot();
  });
});
