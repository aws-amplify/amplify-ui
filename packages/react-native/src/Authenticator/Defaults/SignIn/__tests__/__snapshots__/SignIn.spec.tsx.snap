// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders as expected 1`] = `
Array [
  <View
    accessibilityRole="tablist"
    style={
      Array [
        Object {
          "flexDirection": "row",
          "width": "100%",
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <View
      accessibilityRole="tab"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Array [
            Object {
              "backgroundColor": "#e1e1e1",
              "borderTopColor": "#dcdee0",
              "borderTopWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "padding": 12,
            },
            Object {
              "opacity": 1,
            },
            Object {
              "backgroundColor": "#fff",
              "borderTopColor": "#047d95",
              "color": "#047d95",
            },
            null,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Array [
              Object {
                "color": "#304050",
                "fontSize": 16,
                "fontWeight": "700",
              },
              undefined,
              Object {
                "backgroundColor": "#fff",
                "borderTopColor": "#047d95",
                "color": "#047d95",
              },
            ],
          ]
        }
      >
        Sign In
      </Text>
    </View>
    <View
      accessibilityRole="tab"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Array [
            Object {
              "backgroundColor": "#e1e1e1",
              "borderTopColor": "#dcdee0",
              "borderTopWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "padding": 12,
            },
            null,
            null,
            null,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Array [
              Object {
                "color": "#304050",
                "fontSize": 16,
                "fontWeight": "700",
              },
              undefined,
              null,
            ],
          ]
        }
      >
        Create Account
      </Text>
    </View>
  </View>,
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "black",
        },
        Object {
          "fontSize": 34,
          "fontWeight": "300",
          "lineHeight": 51,
        },
        undefined,
      ]
    }
  />,
  <View>
    <View
      accessibilityLabel="Username"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="Password"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              null,
              null,
            ]
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../../../src/assets/icons/visibilityOff.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 16,
                  "resizeMode": "contain",
                  "tintColor": undefined,
                  "width": 16,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        null,
        Object {
          "backgroundColor": "teal",
          "marginVertical": 8,
          "paddingVertical": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sign in
    </Text>
  </View>,
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Object {
            "marginVertical": 8,
            "paddingVertical": 12,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "color": "teal",
            },
          ]
        }
      >
        Forgot your password?
      </Text>
    </View>
  </View>,
]
`;

exports[`SignIn renders as expected when hideSignUp is true 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "black",
        },
        Object {
          "fontSize": 34,
          "fontWeight": "300",
          "lineHeight": 51,
        },
        undefined,
      ]
    }
  />,
  <View>
    <View
      accessibilityLabel="Username"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="Password"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              null,
              null,
            ]
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../../../src/assets/icons/visibilityOff.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 16,
                  "resizeMode": "contain",
                  "tintColor": undefined,
                  "width": 16,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        null,
        Object {
          "backgroundColor": "teal",
          "marginVertical": 8,
          "paddingVertical": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sign in
    </Text>
  </View>,
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Object {
            "marginVertical": 8,
            "paddingVertical": 12,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "color": "teal",
            },
          ]
        }
      >
        Forgot your password?
      </Text>
    </View>
  </View>,
]
`;

exports[`SignIn renders as expected with an error 1`] = `
Array [
  <View
    accessibilityRole="tablist"
    style={
      Array [
        Object {
          "flexDirection": "row",
          "width": "100%",
        },
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    <View
      accessibilityRole="tab"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Array [
            Object {
              "backgroundColor": "#e1e1e1",
              "borderTopColor": "#dcdee0",
              "borderTopWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "padding": 12,
            },
            Object {
              "opacity": 1,
            },
            Object {
              "backgroundColor": "#fff",
              "borderTopColor": "#047d95",
              "color": "#047d95",
            },
            null,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Array [
              Object {
                "color": "#304050",
                "fontSize": 16,
                "fontWeight": "700",
              },
              undefined,
              Object {
                "backgroundColor": "#fff",
                "borderTopColor": "#047d95",
                "color": "#047d95",
              },
            ],
          ]
        }
      >
        Sign In
      </Text>
    </View>
    <View
      accessibilityRole="tab"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Array [
            Object {
              "backgroundColor": "#e1e1e1",
              "borderTopColor": "#dcdee0",
              "borderTopWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "padding": 12,
            },
            null,
            null,
            null,
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Array [
              Object {
                "color": "#304050",
                "fontSize": 16,
                "fontWeight": "700",
              },
              undefined,
              null,
            ],
          ]
        }
      >
        Create Account
      </Text>
    </View>
  </View>,
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "black",
        },
        Object {
          "fontSize": 34,
          "fontWeight": "300",
          "lineHeight": 51,
        },
        undefined,
      ]
    }
  />,
  <View>
    <View
      accessibilityLabel="Username"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      accessibilityLabel="Password"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          Object {
            "fontSize": 16,
            "marginVertical": 4,
          },
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "marginHorizontal": 4,
              "marginVertical": 2,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            undefined,
          ]
        }
      >
        Password
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "black",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "padding": 8,
            "width": "100%",
          }
        }
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              null,
              null,
            ]
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../../../src/assets/icons/visibilityOff.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 16,
                  "resizeMode": "contain",
                  "tintColor": undefined,
                  "width": 16,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f5bcbc",
          "flexDirection": "row",
          "paddingHorizontal": 4,
          "paddingVertical": 16,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityRole="image"
      source={
        Object {
          "testUri": "../../../src/assets/icons/error.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "resizeMode": "contain",
            "tintColor": undefined,
            "width": 20,
          },
          Object {
            "margin": 4,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 4,
          },
          undefined,
        ]
      }
    >
      An error!
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        null,
        Object {
          "backgroundColor": "teal",
          "marginVertical": 8,
          "paddingVertical": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
          },
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sign in
    </Text>
  </View>,
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          null,
          Object {
            "marginVertical": 8,
            "paddingVertical": 12,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
            },
            Object {
              "color": "teal",
            },
          ]
        }
      >
        Forgot your password?
      </Text>
    </View>
  </View>,
]
`;
